### PR TITLE
Migrate ChatDomain::sendReaction

### LINKED ISSUE #2898: Eliminate ChatDomain::sendReaction
### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Side effects of send reaction are moved to OfflinePlugin
- [ ] Reaction set up is done in UI part, not as part of side effect
- [ ] Calls replaced by ChatClient::sendReaction
- [ ] Method is deprecated

### PR DESCRIPTION
**Pull request by adasiewiczr**
closes #2898  

### 🎯 Goal

Migrate `ChatDomain::sendReaction` to `OfflinePlugin `approach.

### 🛠 Implementation details

1. Created `SendReactionListener ` and OfflinePlugin's implementation
2. Created `SendReactionErrorHandler` and `OfflineErrorHandler` implementation
3. Added `cid: String? = null` to `ChatClient::sendReaction` which is needed to do optimistic reaction update 
4. Removed `ChatClient::sendReaction` without the `Reaction`object 
5. Added a bunch of side effect functions to `ChatClient::sendReaction` method

### 🧪 Testing

Test sending reaction.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] ~Changelog is updated with client-facing changes~
- [x] New code is covered by unit tests
- [ ] ~Comparison screenshots added for visual changes~
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] New feature tested and works
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/plugin/listeners/SendReactionListener.kt
Comment (filbabic) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/plugin/listeners/SendReactionListener.kt:12]: What is the timeline for this API to become non exp?

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt
Comment (leandroBorgesFerreira) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:645]: Please don't create `errorHandler` inside ChatClient as this class should not be aware of how its dependencies are created. We can add the error handlers just like we do with plugins.

**Review by leandroBorgesFerreira — CHANGES_REQUESTED** [review]
Some other comments sent by slack

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/plugin/listeners/SendReactionListener.kt
Comment (adasiewiczr) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/experimental/plugin/listeners/SendReactionListener.kt:12]: After removing ChatDomain

**Review by filbabic — APPROVED** [review]
Works good for me. I'll leave it to the core team to resolve the internal discussions about API design, as Leo has pointed out some good items!

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] New feature tested and works
- [x] All code we touched has new or updated KDocs

**Review by leandroBorgesFerreira — APPROVED** [review]
🚢 🚢 🚢

**Pull request merged**
